### PR TITLE
fix(eslint-config): special case import of "."

### PR DESCRIPTION
**Pull request by wincent**
I don't like these "." imports as there are not very idiomatic; I guess they're not common because they are subtle and potentially confusing. If you are in a folder and "./index" is the manifest of everything exportable in that folder, then that would normally include the file doing the import, which is circular. So the cognitive load of thinking that through is higher than it need be.

Anyway, even though I don't like them, it doesn't mean I shouldn't fix the bug. So here goes.

Closes: https://github.com/liferay/liferay-frontend-projects/issues/129

### CODE REVIEW

**Comment by wincent** [comment]
As I wrote this, I was wondering if a simpler test might suffice, like `source.startsWith('.')`, but I was worried that it would give a false positive if you ever wound up with a package name that started with "." (npm shouldn't allow this, but just in case it ever did... or you created one manually...).

So I renamed a package in a `node_modules` folder to `.crazy` to test this out, and indeed you can see in the REPL that `node` lets you `require` it:

```
> require('.crazy')
{ default: [Function: isObject] }
```

In other words, I think this is the appropriate fix. It's pretty simple and the tests are green so I'm going to merge it. No hurry to cut an `eslint-config` release because of this because the only projects where we're using these weird imports is already right here in the monorepo, where the fix already applies.
